osde2e/cloud_ingress_operator_tests.go

```diff
@@ -102,32 +103,22 @@ var _ = ginkgo.Describe("cloud-ingress-operator", ginkgo.Ordered, func() {
 		cidrBlock := apiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks
 		updatedApiScheme := apiScheme.DeepCopy()
 
-		//reset cidrblock after test is done
-		defer func() {
-			updatedApiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks = apiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks
-			err = k8s.Update(ctx, updatedApiScheme)
-			Expect(err).NotTo(HaveOccurred(), "Could not revert APIScheme CR instance")
-		}()
-
-		// Remove last IP from the cidrBlock:
-		var updatedCidrBlock []string
-		for _, t := range cidrBlock {
-			temp := t[:]
-			updatedCidrBlock = append(updatedCidrBlock, temp)
-		}
-
-		updatedCidrBlock[len(updatedCidrBlock)-1] = ""                // Erase last element (write zero value)
-		updatedCidrBlock = updatedCidrBlock[:len(updatedCidrBlock)-1] // Truncate slice
+		updatedCidrBlock := cidrBlock[:len(cidrBlock)-1]
 
 		// Put the new CIRDBlock ranges into the APIScheme
 		updatedApiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks = updatedCidrBlock
 
 		// Update the APIScheme
 		err = k8s.Update(ctx, updatedApiScheme)
 		Expect(err).NotTo(HaveOccurred(), "Could not update APIScheme CR instance")
-
-		// Wait 30 secs for apiserver to reconcile
-		time.Sleep(30 * time.Second)
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+			if slices.Equal(updatedApiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks, updatedCidrBlock) {
+				log.Println("Updated cidrblock in rh-api service.")
+				return true, nil
+			}
+			return false, nil
+		})
+		Expect(err).NotTo(HaveOccurred(), "Updated cidrblock not able to be set in rh-api service.")
```

Comment:
Removing the arbitrary `sleep` provides more stability for this test and also improves the run time.